magento/magento2
# Trigger page load listeners when no longer loading

### Description (*)
Trigger JavaScript page load listeners to fire when document is no longer 'loading' rather than waiting until it is marked as 'complete' as this can lead to severe delays in rendering JavaScript content.

### Fixed Issues (if relevant)
1. magento/magento2#22909: requirejs/domReady.js can severely delay rendering of content

### Manual testing scenarios (*)
1. Add a non blocking asset to the page with a severe delay e.g. `<img src="https://httpstat.us/404?sleep=10000" />` (I added this to `Magento_Theme::html/absolute_footer.phtml`)
2. Visit the homepage
3. The loading status indicator in the browser should be visible for at least the time set in the `sleep` value of the asset, however private / JavaScript content should render before this finishes (I used the `Default welcome msg!` text in the header to verify this)

### Questions or comments
N/A

### Contribution checklist (*)
